src/components/Comments.jsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from "react";
+import { getCommentByArticleID } from "../api/APICalls";
+import { useParams } from "react-router-dom";
+import Article from "./Article";
+
+export default function Comments(props) {
+  const [comments, setComments] = useState([]);
+  const [singleComment, setSingleComment] = useState({});
+
+  const { article_id } = useParams();
+
+  useEffect(() => {
+    getCommentByArticleID(article_id).then(
+      (response) => {
+        setComments(response);
+      },
+      [article_id]
+    );
+  });
```

Comment:
You are missing the dependency array here, it looks like the above array is in the wrong place. Currently you are making an infinite loop of requests to your backend api as a result of this not being here